api/envoy/extensions/filters/http/rate_limit_quota/v3/rate_limit_quota.proto

```diff
@@ -251,6 +252,13 @@ message RateLimitQuotaBucketSettings {
     // The headers are added even when the rate limit quota was not enforced.
     repeated config.core.v3.HeaderValueOption response_headers_to_add = 4
         [(validate.rules).repeated = {max_items: 10}];
+
+    // Specifies whether a ``RESOURCE_EXHAUSTED`` gRPC code must be returned instead
+    // of the default ``UNAVAILABLE`` gRPC code for a rate limited gRPC call. The
+    // HTTP code will be 200 for a gRPC response.
+    //
+    // This field is ignored if ``grpc_status`` is explicitly set.
+    bool rate_limited_as_resource_exhausted = 5;
```

Comment:
How is setting the `rate_limited_as_resource_exhausted` to `true` different from setting `grpc_status` to `RESOURCE_EXHAUSTED`

Reply:
I see. I think the way the rate_limit filter did it is not as flexible as the rate_limit_quota. Since the behavior is identical I would just use the grpc_status since it is not that much harder.